src/embedded_flash/store/format.rs

```diff
@@ -59,6 +59,14 @@ pub struct Format {
     /// - 1 for insert entries.
     replace_bit: usize,
 
+    /// Whether a user entry has sensitive data.
+    ///
+    /// When a user entry with sensitive data is deleted, the data is overwritten with zeroes.
```

Comment:
Could you be more specific about the contract here? In which cases is data indeed guaranteed to be overwritten at the moment? In which (corruption) cases is it not guaranteed? When moving a sensitive entry for compaction, is the old entry wiped out at an appropriate time? What happens when recovering while some sensitive entry was being copied around?

The contract doesn't have to be very strict for now, but there must at least be a disclaimer about what cases are not covered.

Of course, this contract can evolve as the implementation evolves (better corruption recovery for example).

Reply:
The contract is the same for all operations and documented in the module. Currently, the only failure scenario is power loss with atomic word writes.

Reply:
It would be useful to repeat the assumptions in the documentation of this field and/or at least write a sentence pointing to the module's documentation.

Reply:
Done.